src/main/java/org/fundacionjala/coding/melvi/EANValidator.java

```diff
@@ -0,0 +1,50 @@
+package org.fundacionjala.coding.melvi;
+
+/**
+ *
+ */
+public final class EANValidator {
+
+    public static final int THREE = 3;
+    public static final int TEN = 10;
+
+    /**
+     *
+     */
+    private EANValidator() {
+    }
+
+    /**
+     * @param eanCode code to validate
+     * @return true if valid
+     */
+    public static boolean validate(final String eanCode) {
+        //Your code
+        String code = eanCode;
+        long suma = 0;
+        int last = Integer.parseInt(code.substring(code.length() - 1));
+        for (int i = 0; i < code.length() - 1; i++) {
+            if (i > 0 && i % 2 == 1) {
+                suma += (int) Math.multiplyExact(Integer.parseInt(code.substring(i, i + 1)), THREE);
+
+            }
+            if (i % 2 == 0) {
+                suma += Integer.parseInt(code.substring(i, i + 1));
+            }
+        }
+
+        if (suma % TEN == 0) {
+            if (last == 0) {
+                return true;
+            }
+            return false;
+        } else {
+            long valor = TEN - (suma % TEN);
+            if (last == valor) {
+                return true;
+            }
+            return false;
+        }
```

Comment:
This may be refactored like this.
```java
	public static boolean validate(final String eanCode) {
		//Your eanCode
		int suma = 0;
		for (int i = 0; i < eanCode.length() - 1; i++) {
			int digit = Integer.parseInt(eanCode.substring(i, i + 1));
			suma += i % 2 == 0 ? digit : Math.multiplyExact(digit, THREE);
		}
		int last = Integer.parseInt(eanCode.substring(eanCode.length() - 1));
		long checkSum = suma % TEN == 0 ? 0 : TEN - (suma % TEN);
		return last == checkSum;
	}
```